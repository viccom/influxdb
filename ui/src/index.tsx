import 'babel-polyfill'

import React, {PureComponent} from 'react'
import {render} from 'react-dom'
import {Provider} from 'react-redux'
import {Router, Route, useRouterHistory, IndexRoute} from 'react-router'
import {createHistory, History} from 'history'

import configureStore from 'src/store/configureStore'
import {loadLocalStorage} from 'src/localStorage'

import {getRootNode} from 'src/utils/nodes'
import {getBasepath} from 'src/utils/basepath'

// Components
import App from 'src/App'
import GetSources from 'src/shared/containers/GetSources'
import SetActiveSource from 'src/shared/containers/SetActiveSource'
import GetOrganizations from 'src/shared/containers/GetOrganizations'
import Setup from 'src/Setup'
import Signin from 'src/Signin'
import SigninPage from 'src/onboarding/containers/SigninPage'
import Logout from 'src/Logout'
import TaskPage from 'src/tasks/containers/TaskPage'
import TasksPage from 'src/tasks/containers/TasksPage'
import OrganizationsIndex from 'src/organizations/containers/OrganizationsIndex'
import OrganizationView from 'src/organizations/containers/OrganizationView'
import TaskEditPage from 'src/tasks/containers/TaskEditPage'
import DashboardPage from 'src/dashboards/components/DashboardPage'
import DashboardsIndex from 'src/dashboards/components/dashboard_index/DashboardsIndex'
import DataExplorerPage from 'src/dataExplorer/components/DataExplorerPage'
import {MePage, Account} from 'src/me'
import NotFound from 'src/shared/components/NotFound'
import GetLinks from 'src/shared/containers/GetLinks'
import GetMe from 'src/shared/containers/GetMe'
import SourcesPage from 'src/sources/components/SourcesPage'
import ConfigurationPage from 'src/configuration/components/ConfigurationPage'

import OnboardingWizardPage from 'src/onboarding/containers/OnboardingWizardPage'

// Actions
import {disablePresentationMode} from 'src/shared/actions/app'

// Styles
import 'src/style/chronograf.scss'

const rootNode = getRootNode()
const basepath = getBasepath()

declare global {
  interface Window {
    basepath: string
  }
}

// Older method used for pre-IE 11 compatibility
window.basepath = basepath

const history: History = useRouterHistory(createHistory)({
  basename: basepath, // this is written in when available by the URL prefixer middleware
})

const store = configureStore(loadLocalStorage(), history)
const {dispatch} = store

history.listen(() => {
  dispatch(disablePresentationMode())
})

window.addEventListener('keyup', event => {
  const escapeKeyCode = 27
  // fallback for browsers that don't support event.key
  if (event.key === 'Escape' || event.keyCode === escapeKeyCode) {
    dispatch(disablePresentationMode())
  }
})

class Root extends PureComponent {
  public render() {
    return (
      <Provider store={store}>
        <Router history={history}>
          <Route component={GetLinks}>
            <Route component={Setup}>
              <Route path="/onboarding">
                <Route path=":stepID" component={OnboardingWizardPage} />
                <Route
                  path=":stepID/:substepID"
                  component={OnboardingWizardPage}
                />
              </Route>
              <Route path="/signin" component={SigninPage} />
              <Route path="/logout" component={Logout} />
              <Route path="/">
                <Route component={Signin}>
                  <Route component={GetMe}>
                    <Route component={GetOrganizations}>
                      <Route component={App}>
                        <Route component={GetSources}>
                          <Route component={SetActiveSource}>
                            <IndexRoute component={DashboardsIndex} />
                            <Route path="organizations">
                              <IndexRoute component={OrganizationsIndex} />
                              <Route
                                path=":orgID/:tab"
                                component={OrganizationView}
                              />
                            </Route>
                            <Route path="tasks">
                              <IndexRoute component={TasksPage} />
                              <Route path="new" component={TaskPage} />
                              <Route path=":id" component={TaskEditPage} />
                            </Route>
                            <Route
                              path="data-explorer"
                              component={DataExplorerPage}
                            />
                            <Route path="dashboards">
                              <IndexRoute component={DashboardsIndex} />
                              <Route
                                path=":dashboardID"
                                component={DashboardPage}
                              />
                            </Route>
                            <Route path="me" component={MePage} />
                            <Route path="account/:tab" component={Account} />
                            <Route path="sources" component={SourcesPage} />
                            <Route
                              path="configuration/:tab"
                              component={ConfigurationPage}
                            />
                          </Route>
                        </Route>
                      </Route>
                    </Route>
                  </Route>
                </Route>
              </Route>
            </Route>
          </Route>
          <Route path="*" component={NotFound} />
        </Router>
      </Provider>
    )
  }
}

if (rootNode) {
  render(<Root />, rootNode)
}
