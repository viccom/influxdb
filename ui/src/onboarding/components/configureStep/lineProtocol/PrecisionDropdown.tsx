// Libraries
import React, {PureComponent} from 'react'

// Components
import {Dropdown} from 'src/clockface'

// Types
import {WritePrecision} from 'src/api'
import {Precision} from 'src/types/v2/dataLoaders'

interface Props {
  setPrecision: (precision: WritePrecision) => void
  precision: WritePrecision
}

const writePrecisions: WritePrecision[] = [
  WritePrecision.Ns,
  WritePrecision.Us,
  WritePrecision.Ms,
  WritePrecision.S,
]

const makePrecisionReadable = {
  [WritePrecision.Ns]: Precision.Nanoseconds,
  [WritePrecision.Us]: Precision.Microseconds,
  [WritePrecision.S]: Precision.Seconds,
  [WritePrecision.Ms]: Precision.Milliseconds,
}

class PrecisionDropdown extends PureComponent<Props> {
  public render() {
    const {setPrecision, precision} = this.props
    return (
      <div className={'wizard-step--footer dropdown'}>
        <label>Time Precision </label>
        <Dropdown
          selectedID={precision}
          onChange={setPrecision}
          widthPixels={200}
        >
          {writePrecisions.map(value => (
            <Dropdown.Item key={value} value={value} id={value}>
              {makePrecisionReadable[value]}
            </Dropdown.Item>
          ))}
        </Dropdown>
      </div>
    )
  }
}

export default PrecisionDropdown
