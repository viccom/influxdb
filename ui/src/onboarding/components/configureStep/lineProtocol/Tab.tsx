import React, {PureComponent} from 'react'

import {Radio} from 'src/clockface'

import {LineProtocolTab} from 'src/types/v2/dataLoaders'

interface Props {
  active: boolean
  tab: LineProtocolTab
  onClick: (tab: LineProtocolTab) => void
}

export default class extends PureComponent<Props> {
  public render() {
    const {tab, active} = this.props

    return (
      <Radio.Button
        key={tab}
        id={tab}
        titleText={tab}
        value={tab}
        active={active}
        onClick={this.handleClick}
      >
        {tab}
      </Radio.Button>
    )
  }

  private handleClick = () => {
    this.props.onClick(this.props.tab)
  }
}
