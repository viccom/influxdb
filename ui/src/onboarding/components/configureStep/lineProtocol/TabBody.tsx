import React, {PureComponent, ChangeEvent} from 'react'

import {
  Input,
  InputType,
  Form,
  Grid,
  Columns,
  ComponentSize,
} from 'src/clockface'
import DragAndDrop from 'src/shared/components/DragAndDrop'
import TextArea from 'src/clockface/components/inputs/TextArea'
import {LineProtocolTab} from 'src/types/v2/dataLoaders'
import {setLineProtocolBody} from 'src/onboarding/actions/dataLoaders'
import {WritePrecision} from 'src/api'

interface Props {
  lineProtocolBody: string
  activeLPTab: LineProtocolTab
  precision: WritePrecision
  setLineProtocolBody: typeof setLineProtocolBody
  onURLChange: (url: string) => void
  urlInput: string
  handleSubmit?: () => void
}

export default class extends PureComponent<Props> {
  public render() {
    const {lineProtocolBody, activeLPTab, urlInput} = this.props

    switch (activeLPTab) {
      case LineProtocolTab.UploadFile:
        return (
          <DragAndDrop
            submitText="Upload File"
            handleSubmit={this.handleSetLineProtocol}
            submitOnDrop={true}
            submitOnUpload={true}
          />
        )
      case LineProtocolTab.EnterManually:
        return (
          <TextArea
            value={lineProtocolBody}
            placeholder="Write text here"
            onChange={this.handleSetLineProtocol}
          />
        )
      case LineProtocolTab.EnterURL:
        return (
          <Grid>
            <Grid.Row>
              <Grid.Column
                widthXS={Columns.Twelve}
                widthMD={Columns.Ten}
                offsetMD={Columns.One}
              >
                <Form.Element label="File URL:">
                  <Input
                    titleText="File URL:"
                    type={InputType.Text}
                    placeholder="http://..."
                    value={urlInput}
                    onChange={this.handleChange}
                    autoFocus={true}
                    size={ComponentSize.Large}
                  />
                </Form.Element>
              </Grid.Column>
            </Grid.Row>
          </Grid>
        )
    }
  }

  private handleChange = (e: ChangeEvent<HTMLInputElement>) => {
    const {value} = e.target
    this.props.onURLChange(value)
  }

  private handleSetLineProtocol = async (lpBody: string) => {
    const {setLineProtocolBody, handleSubmit} = this.props
    await setLineProtocolBody(lpBody)
    if (handleSubmit) {
      handleSubmit()
    }
  }
}
