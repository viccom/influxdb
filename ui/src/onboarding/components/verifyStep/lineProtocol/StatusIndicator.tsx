// Libraries
import React, {PureComponent} from 'react'
import classnames from 'classnames'

import {SparkleSpinner} from 'src/clockface'

// Types
import {RemoteDataState} from 'src/types'

interface Props {
  status: RemoteDataState
  onClickRetry: () => void
}

class StatusIndicator extends PureComponent<Props> {
  public render() {
    const {status} = this.props
    return (
      <>
        <div className={'wizard-step--top-container'}>
          <div className={'wizard-step--sparkle-container'}>
            <SparkleSpinner loading={status} />
          </div>
        </div>
        <div className={'wizard-step--footer'}>
          <div className={this.footerClass}>{this.footerText}</div>
        </div>
        <br />
      </>
    )
  }

  private get footerClass(): string {
    const {status} = this.props

    return classnames(`wizard-step--text-state`, {
      loading: status === RemoteDataState.Loading,
      success: status === RemoteDataState.Done,
      error: status === RemoteDataState.Error,
    })
  }

  private get footerText(): string {
    switch (this.props.status) {
      case RemoteDataState.Loading:
        return 'Loading...'
      case RemoteDataState.Done:
        return 'Data Written Successfully!'
      case RemoteDataState.Error:
        return 'Unable to Write Data'
    }
  }
}

export default StatusIndicator
