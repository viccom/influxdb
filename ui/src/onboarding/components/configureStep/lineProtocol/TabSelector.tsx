import React, {PureComponent} from 'react'

import {Radio, ButtonShape} from 'src/clockface'
import {LineProtocolTab} from 'src/types/v2/dataLoaders'

import Tab from 'src/onboarding/components/configureStep/lineProtocol/Tab'

interface Props {
  tabs: LineProtocolTab[]
  activeLPTab: LineProtocolTab
  onClick: (tab: LineProtocolTab) => void
}

export default class extends PureComponent<Props> {
  public render() {
    const {tabs, activeLPTab} = this.props
    return (
      <Radio shape={ButtonShape.Default}>
        {tabs.map(t => (
          <Tab
            tab={t}
            key={t}
            active={activeLPTab === t}
            onClick={this.handleTabClick}
          />
        ))}
      </Radio>
    )
  }

  private handleTabClick = (tab: LineProtocolTab) => {
    const {activeLPTab, onClick} = this.props
    if (tab !== activeLPTab) {
      onClick(tab)
    }
  }
}
