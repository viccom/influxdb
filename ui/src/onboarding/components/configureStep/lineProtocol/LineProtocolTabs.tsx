// Libraries
import React, {PureComponent} from 'react'
import {connect} from 'react-redux'
import _ from 'lodash'

import PrecisionDropdown from 'src/onboarding/components/configureStep/lineProtocol/PrecisionDropdown'
import TabSelector from 'src/onboarding/components/configureStep/lineProtocol/TabSelector'
import TabBody from 'src/onboarding/components/configureStep/lineProtocol/TabBody'

// Types
import {LineProtocolTab} from 'src/types/v2/dataLoaders'

// Components
import {Grid, Columns} from 'src/clockface'

// Actions
import {
  setLineProtocolBody,
  setActiveLPTab,
  writeLineProtocolAction,
  setPrecision,
} from 'src/onboarding/actions/dataLoaders'

import {AppState} from 'src/types/v2/index'

// Styles
import 'src/clockface/components/auto_input/AutoInput.scss'
import {WritePrecision} from 'src/api'

interface OwnProps {
  tabs: LineProtocolTab[]
  bucket: string
  org: string
  handleSubmit?: () => void
}

type Props = OwnProps & DispatchProps & StateProps

interface DispatchProps {
  setLineProtocolBody: typeof setLineProtocolBody
  setActiveLPTab: typeof setActiveLPTab
  writeLineProtocolAction: typeof writeLineProtocolAction
  setPrecision: typeof setPrecision
}

interface StateProps {
  lineProtocolBody: string
  activeLPTab: LineProtocolTab
  precision: WritePrecision
}

interface State {
  urlInput: string
}

export class LineProtocolTabs extends PureComponent<Props, State> {
  constructor(props: Props) {
    super(props)

    this.state = {
      urlInput: '',
    }
  }

  public render() {
    const {
      setPrecision,
      precision,
      activeLPTab,
      tabs,
      setLineProtocolBody,
      lineProtocolBody,
      handleSubmit,
    } = this.props

    const {urlInput} = this.state

    return (
      <div>
        <TabSelector
          activeLPTab={activeLPTab}
          tabs={tabs}
          onClick={this.handleTabClick}
        />

        <Grid>
          <Grid.Row>
            <Grid.Column
              widthXS={Columns.Twelve}
              widthMD={Columns.Ten}
              offsetMD={Columns.One}
              widthLG={Columns.Eight}
              offsetLG={Columns.Two}
            >
              <div className="onboarding--admin-user-form">
                <div className={'wizard-step--lp-body'}>
                  <TabBody
                    onURLChange={this.handleURLChange}
                    activeLPTab={activeLPTab}
                    precision={precision}
                    urlInput={urlInput}
                    lineProtocolBody={lineProtocolBody}
                    setLineProtocolBody={setLineProtocolBody}
                    handleSubmit={handleSubmit}
                  />
                </div>

                <PrecisionDropdown
                  setPrecision={setPrecision}
                  precision={precision}
                />
              </div>
            </Grid.Column>
          </Grid.Row>
        </Grid>
      </div>
    )
  }

  private handleTabClick = (tab: LineProtocolTab) => {
    const {setActiveLPTab, setLineProtocolBody} = this.props

    setLineProtocolBody('')
    setActiveLPTab(tab)
  }

  private handleURLChange = (urlInput: string) => {
    this.setState({urlInput})
  }
}

const mstp = ({
  dataLoading: {
    dataLoaders: {lineProtocolBody, activeLPTab, precision},
  },
}: AppState) => {
  return {lineProtocolBody, activeLPTab, precision}
}

const mdtp: DispatchProps = {
  setLineProtocolBody,
  setActiveLPTab,
  writeLineProtocolAction,
  setPrecision,
}

export default connect<StateProps, DispatchProps, OwnProps>(
  mstp,
  mdtp
)(LineProtocolTabs)
