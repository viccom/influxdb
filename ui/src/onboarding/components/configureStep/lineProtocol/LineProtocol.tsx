// Libraries
import React, {PureComponent} from 'react'
import {connect} from 'react-redux'

import _ from 'lodash'

// Components
import LineProtocolTabs from 'src/onboarding/components/configureStep/lineProtocol/LineProtocolTabs'
import OnboardingButtons from 'src/onboarding/components/OnboardingButtons'
import {Form} from 'src/clockface'
import FancyScrollbar from 'src/shared/components/fancy_scrollbar/FancyScrollbar'

// Actions
import {
  setLPStatus as setLPStatusAction,
  writeLineProtocolAction,
} from 'src/onboarding/actions/dataLoaders'

// Decorator
import {ErrorHandling} from 'src/shared/decorators/errors'

// Types
import {LineProtocolTab} from 'src/types/v2/dataLoaders'
import {AppState} from 'src/types/v2/index'
import {WritePrecision} from 'src/api'
import {RemoteDataState} from 'src/types'

interface OwnProps {
  bucket: string
  org: string
  onClickNext: () => void
  onClickBack: () => void
  onClickSkip: () => void
}

interface StateProps {
  lineProtocolBody: string
  precision: WritePrecision
}

interface DispatchProps {
  setLPStatus: typeof setLPStatusAction
  writeLineProtocolAction: typeof writeLineProtocolAction
}

type Props = OwnProps & StateProps & DispatchProps

@ErrorHandling
export class LineProtocol extends PureComponent<Props> {
  public componentDidMount() {
    const {setLPStatus} = this.props
    setLPStatus(RemoteDataState.NotStarted)
  }

  public render() {
    return (
      <Form onSubmit={this.handleSubmit}>
        <div className="wizard-step--scroll-area">
          <FancyScrollbar autoHide={false}>
            <div className="wizard-step--scroll-content">
              <h3 className="wizard-step--title">Add Data via Line Protocol</h3>
              <h5 className="wizard-step--sub-title">
                Need help writing InfluxDB Line Protocol? See Documentation
              </h5>
              {this.Content}
            </div>
          </FancyScrollbar>
        </div>
        <OnboardingButtons
          onClickBack={this.props.onClickBack}
          onClickSkip={this.props.onClickSkip}
          showSkip={true}
          autoFocusNext={true}
          skipButtonText={'Skip Config'}
        />
      </Form>
    )
  }

  private get LineProtocolTabs(): LineProtocolTab[] {
    return [LineProtocolTab.UploadFile, LineProtocolTab.EnterManually]
  }

  private get Content(): JSX.Element {
    const {bucket, org} = this.props
    return (
      <LineProtocolTabs
        tabs={this.LineProtocolTabs}
        bucket={bucket}
        org={org}
        handleSubmit={this.handleSubmit}
      />
    )
  }

  private handleSubmit = () => {
    const {
      bucket,
      org,
      writeLineProtocolAction,
      lineProtocolBody,
      precision,
    } = this.props

    writeLineProtocolAction(org, bucket, lineProtocolBody, precision)
    this.props.onClickNext()
  }
}

const mstp = ({
  dataLoading: {
    dataLoaders: {lineProtocolBody, precision},
  },
}: AppState): StateProps => {
  return {lineProtocolBody, precision}
}

const mdtp: DispatchProps = {
  setLPStatus: setLPStatusAction,
  writeLineProtocolAction,
}

export default connect<StateProps, DispatchProps, OwnProps>(
  mstp,
  mdtp
)(LineProtocol)
