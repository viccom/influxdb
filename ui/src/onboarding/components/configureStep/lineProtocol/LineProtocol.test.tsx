// Libraries
import React from 'react'
import {shallow} from 'enzyme'

// Components
import {LineProtocol} from 'src/onboarding/components/configureStep/lineProtocol/LineProtocol'
import {WritePrecision} from 'src/api'

const setup = (override = {}) => {
  const props = {
    bucket: 'a',
    org: 'a',
    onClickNext: jest.fn(),
    onClickBack: jest.fn(),
    onClickSkip: jest.fn(),
    lineProtocolBody: '',
    precision: WritePrecision.Ns,
    setLPStatus: jest.fn(),
    writeLineProtocolAction: jest.fn(),
    ...override,
  }
  const wrapper = shallow(<LineProtocol {...props} />)

  return {wrapper}
}

describe('LineProtocol', () => {
  describe('rendering', () => {
    it('renders!', () => {
      const {wrapper} = setup()
      expect(wrapper.exists()).toBe(true)

      expect(wrapper).toMatchSnapshot()
    })
  })
})
