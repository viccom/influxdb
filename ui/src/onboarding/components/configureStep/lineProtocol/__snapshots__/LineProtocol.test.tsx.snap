// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LineProtocol rendering renders! 1`] = `
<Form
  onSubmit={[Function]}
>
  <div
    className="wizard-step--scroll-area"
  >
    <FancyScrollbar
      autoHeight={false}
      autoHide={false}
      maxHeight={null}
      setScrollTop={[Function]}
      style={Object {}}
    >
      <div
        className="wizard-step--scroll-content"
      >
        <h3
          className="wizard-step--title"
        >
          Add Data via Line Protocol
        </h3>
        <h5
          className="wizard-step--sub-title"
        >
          Need help writing InfluxDB Line Protocol? See Documentation
        </h5>
        <Connect(LineProtocolTabs)
          bucket="a"
          handleSubmit={[Function]}
          org="a"
          tabs={
            Array [
              "Upload File",
              "Enter Manually",
            ]
          }
        />
      </div>
    </FancyScrollbar>
  </div>
  <OnboardingButtons
    autoFocusNext={true}
    backButtonText="Previous"
    nextButtonStatus="default"
    nextButtonText="Continue"
    onClickBack={[MockFunction]}
    onClickSkip={[MockFunction]}
    showSkip={true}
    skipButtonText="Skip Config"
  />
</Form>
`;
